assignment/assignment.hs

```diff
@@ -26,7 +27,11 @@ import Data.Char(intToDigit)
 -- predicate returns `True`. Implement `myfilter` using `foldl`. (Hint: you 
 -- might need to use `++` rather than `:` to build the result)
 
--- myfilter :: (a -> Bool) -> [a] -> [a]
+--this is too hard or i'm just dumb
+--myfilter :: (a -> Bool) -> [a] -> [a]
+--myfilter pred = foldl (pred && (++)) []
+  -- | pred = foldl (flip (++)) []
+  -- | otherwise =  -- i couldn't figure out how to use an if-then statement WITHOUT an else clause so I found "when"
 
```

Comment:
I suspect you're actually trying to be too clever :)

Nested folds are a path to madness, and shouldn't be necessary as you only need to traverse the list once.

My solution was:

~~~ hs
myfilter :: (a -> Bool) -> [a] -> [a]
myfilter predicate  =
  foldl tester [] 
  where
    tester result value
      | predicate value = result ++ [ value ]
      | otherwise       = result
~~~